nf_core/modules.py

```diff
@@ -231,3 +236,325 @@ def has_valid_pipeline(self):
+                # main.nf
+                with open(os.path.join(test_dir, "main.nf"), "w") as fh:
+                    fh.write(test_nf)
+                # test.yml
+                with open(os.path.join(test_dir, "test.yml"), "w") as fh:
+                    fh.write(test_yml)
+            except OSError as e:
+                log.error(f"Could not create module files: {e}")
+                return False
+
+            # Add line to filters.yml
+            try:
+                with open(os.path.join(directory, ".github", "filters.yml"), "a") as fh:
+                    if subtool:
+                        content = (
+                            "\n"
+                            + f"{tool_name}:"
+                            + "\n"
+                            + f"  - software/{tool}/{subtool}/**"
+                            + "\n"
+                            + f"  - tests/software/{tool}/{subtool}/**\n"
+                        )
+                    else:
+                        content = (
+                            "\n"
+                            + f"{tool_name}:"
+                            + "\n"
+                            + f"  - software/{tool}/**"
+                            + "\n"
+                            + f"  - tests/software/{tool}/**\n"
+                        )
+                    fh.write(content)
+
+            except FileNotFoundError as e:
+                log.error(f"Could not open filters.yml file!")
+                return False
+
+            log.info(f"Successfully created module files at: {tool_dir}")
+            log.info(f"Added test files at: {test_dir}")
+            return True
+
+    def get_repo_type(self, directory):
+        """
+        Determine whether this is a pipeline repository or a clone of
+        nf-core/modules
+        """
+        # Verify that the pipeline dir exists
+        if dir is None or not os.path.exists(directory):
+            log.error("Could not find directory: {}".format(directory))
+            sys.exit(1)
+
+        # Determine repository type
+        if os.path.exists(os.path.join(directory, "main.nf")):
+            return "pipeline"
+        elif os.path.exists(os.path.join(directory, "software")):
+            return "modules"
+        else:
+            log.error("Could not determine repository type of {}".format(directory))
+            sys.exit(1)
+
+    def download_template(self, url):
+        """ Download the module template """
+
+        r = requests.get(url=url)
+
+        if r.status_code != 200:
+            log.error("Could not download the template")
+            sys.exit(1)
+        else:
+            try:
+                template_copy = r.content.decode("ascii")
+            except UnicodeDecodeError as e:
+                log.error(f"Could not decode template file from {url}: {e}")
+                sys.exit(1)
+
+        return template_copy
+
+
+def _bioconda_package(package, full_dep=True):
```

Comment:
This is basically a cut down version of the linting code right?

https://github.com/nf-core/tools/blob/cae0402226983ebaf186a64fe5f1d3a17363f69d/nf_core/lint/conda_env_yaml.py#L184-L230 

I wonder if we make a version of the function that works for both and stick that into `utils.py`..

Reply:
yes, it's also used by `modules lint` 👍